src/services/services.ts

```diff
@@ -1158,6 +1163,86 @@ namespace ts {
+        getSemanticDiagnostics?(fileName: string): Diagnostic[];
+        getEncodedSyntacticClassifications?(fileName: string, span: TextSpan): Classifications;
+        getEncodedSemanticClassifications?(fileName: string, span: TextSpan): Classifications;
+        getCompletionsAtPosition?(fileName: string, position: number): CompletionInfo;
+        getCompletionEntryDetails?(fileName: string, position: number, entryName: string): CompletionEntryDetails;
+        getQuickInfoAtPosition?(fileName: string, position: number): QuickInfo;
+        getNameOrDottedNameSpan?(fileName: string, startPos: number, endPos: number): TextSpan;
+        getBreakpointStatementAtPosition?(fileName: string, position: number): TextSpan;
+        getSignatureHelpItems?(fileName: string, position: number): SignatureHelpItems;
+        getRenameInfo?(fileName: string, position: number): RenameInfo;
+        findRenameLocations?(fileName: string, position: number, findInStrings: boolean, findInComments: boolean): RenameLocation[];
+        getDefinitionAtPosition?(fileName: string, position: number): DefinitionInfo[];
+        getTypeDefinitionAtPosition?(fileName: string, position: number): DefinitionInfo[];
+        getReferencesAtPosition?(fileName: string, position: number): ReferenceEntry[];
+        findReferences?(fileName: string, position: number): ReferencedSymbol[];
+        getDocumentHighlights?(fileName: string, position: number, filesToSearch: string[]): DocumentHighlights[];
+        getNavigateToItems?(searchValue: string, maxResultCount: number): NavigateToItem[];
+        getNavigationBarItems?(fileName: string): NavigationBarItem[];
+        getOutliningSpans?(fileName: string): OutliningSpan[];
+        getTodoComments?(fileName: string, descriptors: TodoCommentDescriptor[]): TodoComment[];
+        getBraceMatchingAtPosition?(fileName: string, position: number): TextSpan[];
+        getIndentationAtPosition?(fileName: string, position: number, options: EditorOptions): number;
+        getFormattingEditsForRange?(fileName: string, start: number, end: number, options: FormatCodeOptions): TextChange[];
+        getFormattingEditsForDocument?(fileName: string, options: FormatCodeOptions): TextChange[];
+        getFormattingEditsAfterKeystroke?(fileName: string, position: number, key: string, options: FormatCodeOptions): TextChange[];
+        getDocCommentTemplateAtPosition?(fileName: string, position: number): TextInsertion;
+
+        // Filters
+
+        // A plugin can implement one of the filter methods to augment, extend or modify a result
+        // prior to the host receiving it. The TypeScript language service is invoked and the
+        // result is passed to the plugin as the value of the previous parameter. If more than one
+        // plugin is registered, the plugins are consulted in the order they are returned from the
+        // host. The value passed in as previous is the result returned by the prior plugin. If a

```

Comment:
Consider replacing `host` with `Program.getCompilerExtensions()` as my proposal had the language service host returning the extensions but this PR gets them from the program.